pandas/core/resample.py

```diff
@@ -1413,6 +1413,16 @@ def _get_time_bins(self, ax):
                                      ambiguous='infer',
                                      nonexistent='shift_forward')
 
+        # GH #24972
+        # In edge case of tz-aware grouping binner last index can be
+        # less than the ax.max() variable in data object, this happens
```

Comment:
should go in _adjust_bin_edges

Reply:
there are a lot of issues with generating the remaining bins when I try to move it there. if this way of generating the bins is the correct one then I believe that it is better to leave it here. in _adjust_bin_edges we lose information about ax tz

Reply:
Or this should possibly be fixed in `_get_timestamp_range_edges`. This patch here seems like a bandaid.

Reply:
ok, let me apply the patch you suggested in the issue, it does look better for me.